kong/plugins/method-restriction/schema.lua

```diff
@@ -0,0 +1,7 @@
+local ALLOWED_METHODS = { "GET", "POST", "PUT", "DELETE", "OPTIONS", "HEAD" }
+
+return {
+  fields = {
+    methods = { type = "array", enum = ALLOWED_METHODS }
```

Comment:
- This field should be called `allowed_methods` or `blocked_methods`. Reading this configuration I cannot be sure if setting a method will mean that it will be whitelisted or blacklisted by the plugin.
- I assume this is currently the equivalent of `blocked_methods`, but a whitelisted, `allowed_methods` field could be useful too.
- Thirdly, the enum is missing the `CONNECT` and `TRACE` HTTP methods as described by HTTP/1.1 RFC
- Last but not least: Kong's goal is to be as less opinionated as possible regarding the specifications it implements. As such, I would actually *not* limit the possible values for this property with an enum, as it is block eventual custom methods that some users might be using with their APIs. Removing the `enum` field and doing case-insensitive comparison should be enough.